### PR TITLE
financial#121: Support modifying Soft Credit permissions via hook

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
financial#121: Support modifying Soft Credit permissions via hook

This is https://github.com/civicrm/civicrm-core/pull/21065 updated to
- remove merge conflict
- get tests passing (cross fingers)
- put tests in the 'right place' - the extension

@MegaphoneJon I did observe the pcp details might not show up for some users - we might need to add pcp to the list of entities that has the same permissions as contributions do

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
I think this is a cleanup issue as they pass locally I think

Test Result (3 failures / +3)
Civi\Financialacls\LineItemTest.testLineItemApiPermissions with data set "APIv3"
Civi\Financialacls\LineItemTest.testLineItemApiPermissions with data set "APIv4"
CRM_Contribute_Page_AjaxTest.testGetSoftContributionSelector

**Comment by eileenmcnaughton** [comment]
test this please

File: CRM/Contribute/BAO/ContributionSoft.php [hunk 1]
```diff
@@ -410,91 +408,64 @@ public static function getSoftContributionList($contact_id, $filter = NULL, $isT
...
-    if (!empty($dTParams['sort'])) {
-      $orderBy = $dTParams['sort'];
+
+    $contributionSofts = ContributionSoft::get()
+      ->addSelect('*', 'contribution_id.receive_date', 'contribution_id.contact_id', 'contribution_id.contact_id.display_name', 'soft_credit_type_id:label', 'contribution_id.contribution_status_id:label', 'contribution_id.financial_type_id:label', 'pcp_id.title')
+      ->addWhere('contact_id', '=', $contact_id)
+      ->addWhere('contribution_id.is_test', '=', $isTest);
+
+    if ($membershipID) {
+      $contributionSofts->addJoin('LineItem', 'INNER', NULL,
```
Comment (seamuslee001) [inline CRM/Contribute/BAO/ContributionSoft.php:418]: @eileenmcnaughton I know that this is the right direction but I'm just wondering if we need to do a status check / upgrade to check that every membership - contribution linkage in civicrm_membership_payment has a correct line item record?

Comment (eileenmcnaughton) [reply CRM/Contribute/BAO/ContributionSoft.php:418]: @seamuslee001 maybe but not specifically for this PR I don't think - there are other places in the code that expect valid data in lineitems

**Comment by eileenmcnaughton** [comment]
@MegaphoneJon I'm wondering if you can re-test this over the 5.48rc (or on master) - I think there is a patch merged for the issue you hit last time - which I believe was because CiviMember was disabled.

**Comment by eileenmcnaughton** [comment]
@MegaphoneJon I just rebased this - would be good if you could test - I think prior issues were resolved outside this pr

**Comment by eileenmcnaughton** [comment]
@MegaphoneJon - just a reminder - I think this should be good to merge - but needs your review

**Comment by MegaphoneJon** [comment]
I had actually completed code review on this yesterday and was going to do manual testing today or Monday.  I'm surprised to see this last commit and don't really understand where the `$additionalWhere` clause was originally?

**Comment by eileenmcnaughton** [comment]
@MegaphoneJon Argh - I just rebased & pushed to keep it all fresh - did I change it?

**Comment by eileenmcnaughton** [comment]
@MegaphoneJon is this still on your todo list?

**Comment by MegaphoneJon** [comment]
I'll put it on the todo list of @briennekordis.

**Comment by eileenmcnaughton** [comment]
thanks -  let's get jenkins to test again - test this please

**Comment by briennekordis** [comment]
@eileenmcnaughton I manually tested this patch on a local dev site and it does modify the permissions of soft credits via the hook, so this patch is good to merge!

**Comment by eileenmcnaughton** [comment]
yay

**Comment by eileenmcnaughton** [comment]
thanks @briennekordis

**Comment by eileenmcnaughton** [comment]
Minor related regression https://github.com/civicrm/civicrm-core/pull/25460

**Pull request merged**